include/ginkgo/core/reorder/reordering_base.hpp

```diff
@@ -60,6 +60,12 @@ namespace reorder {
  * specific reordering to decide what to do with the data that is passed to it.
  */
 class ReorderingBase : public EnableAbstractPolymorphicObject<ReorderingBase> {
+public:
+    virtual std::shared_ptr<const gko::LinOp> get_permutation() const = 0;
+
+    virtual std::shared_ptr<const gko::LinOp> get_inverse_permutation()
+        const = 0;
```

Comment:
Is it necessary to have these getters here ?

Reply:
As the scaled reordered factory takes a `ReorderingBaseFactory`, I think these are necessary here rather than only in the algorithm implemenations.